nmrfx-processor/src/main/java/org/nmrfx/processor/datasets/vendor/NMRDataUtil.java

```diff
@@ -437,6 +437,25 @@ public static String calculateHash(String input) throws NoSuchAlgorithmException
         return base64.encodeToString(digest.digest());
     }
 
+    /**
+     * Attempt to get the phases from the NMRData object, if the phases are not present or are both zero, then
+     * the phases from an auto phase are returned.
+     * @param nmrData The NMRData to retrieve phases from
+     * @param dim The dimension to get the phases for
+     * @return An array of phases
+     */
+    public static double[] getPhases(NMRData nmrData, int dim) {
+        double[] phases = new double[2];
+        if (nmrData.arePhasesSet(dim)) {
+            phases[0] = nmrData.getPH0(dim);
+            phases[1] = nmrData.getPH1(dim);
+        } else {
+            log.info("Getting phases using autophase.");
+            phases = autoPhase(nmrData);
+        }
```

Comment:
autoPhase doesn't have any reference to dimensions so not sure if this is okay to use for all dimensions?

Reply:
Don't use autophase if dimension is greater than 0